lib/google_contacts_api/contacts_accessor.rb

```diff
@@ -25,5 +25,37 @@ def get_contacts(params = {})
       end
       GoogleContactsApi::ContactSet.new(response.body, @api)
     end
+
+    def post_contact(contact)
+      params = {}
+      headers = {
+          'Content-type' => 'application/json'
+          }
+      body = contact.json
+      params[:body] = body
+      url = "contacts/default/full"
+      response = @api.post_v2(url, params, headers)
+      GoogleContactsApi::Api.parse_response_code(response)
+    end
+
+    def put_contact(contact)
+      params = {}
+      headers = {
+          'Content-type' => 'application/json',
+          'If-Match' => '*'
+          }
+      body = contact.json
+      params[:body] = body
+      response = @api.put_v2(contact.edit_link, params, headers)
+      GoogleContactsApi::Api.parse_response_code(response)
+    end
+
+    def put_or_post_contact(contact)
```

Comment:
This is doing the high level logic that tries to do a put and then does a post if that fails.  I figured since the get_contacts method that was already there handled some api response codes, it was better to put that logic in this class and remove that level of detail from the user class?